core/src/main/java/org/infinispan/remoting/inboundhandler/BaseBlockingRunnable.java

```diff
@@ -28,10 +28,18 @@ protected BaseBlockingRunnable(BasePerCacheInboundInvocationHandler handler, Cac
 
    @Override
    public void run() {
+      InvocationStatus invocationStatus = null;
       try {
-         response = beforeInvoke();
-         if (response == null) {
-            response = handler.invokePerform(command);
+         invocationStatus = beforeInvoke();
+         switch (invocationStatus) {
+            case CONTINUE:
+               response = handler.invokePerform(command);
+               break;
+            case CACHE_NOT_FOUND:
+               response = CacheNotFoundResponse.INSTANCE;
+               break;
+            case DEFERRED:
+               return;
```

Comment:
@belaban I know it's not the way it works *now*. I got that you don't want to consider this as an option - and TBH it makes more sense ordering any RPC per-key directly in Infinispan, rather than global ordering within JGroups. It's just that some parts in Infinispan still rely on ordering by JGroups. I silently hope that when Dan finishes his _opus magnum interceptorae_ the async modes will be removed, too.